src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx

```diff
@@ -148,7 +146,55 @@ class OrganizationIntegrations extends AsyncComponent {
           onReinstall={this.onInstall}
           enabledPlugins={this.enabledPlugins}
         />
-      ));
+      </IntegrationRow>
+    );
+  }
+
+  renderSentryApps(apps, key) {
+    const {appInstalls} = this.state;
+
+    return (
+      <IntegrationRow key={`row-${key}`}>
+        <SentryAppInstallations
+          key={key}
+          orgId={this.props.params.orgId}
+          installs={appInstalls}
+          applications={apps}
+        />
+      </IntegrationRow>
+    );
+  }
+
+  renderBody() {
+    const {reloading, applications, appInstalls} = this.state;
+
+    const installedProviders = this.providers
+      .filter(p => p.isInstalled)
+      .map(p => [p.name, this.renderProvider(p)]);
+
+    const uninstalledProviders = this.providers
+      .filter(p => !p.isInstalled)
+      .map(p => [p.name, this.renderProvider(p)]);
+
+    const installedSentryApps = (applications || [])
+      .filter(a => appInstalls.find(i => i.app.slug === a.slug))
+      .map(a => [a.name, this.renderSentryApps([a], a.slug)]);
+
+    const uninstalledSentryApps = (applications || [])
+      .filter(a => !appInstalls.find(i => i.app.slug === a.slug))
+      .map(a => [a.name, this.renderSentryApps([a], a.slug)]);
+
+    // Combine the list of Providers and Sentry Apps that have installations.
+    const installed = installedProviders
+      .concat(installedSentryApps)
+      .sort((a, b) => a[0].localeCompare(b[0]))
```

Comment:
imo, destructuring adds some more code but can be a bit more clear about what you're retrieving from the array:

```
const installed = installedProviders
      .concat(installedSentryApps)
      .sort(([name1,], [name2,]) => name1.localeCompare(name2))
      .map([,Component] => Component);
```

Reply:
Yea, more "punctuation", but definitely clearer 👍